src/components/layout/header/platform-switcher/PlatformSwitcher.tsx

```diff
@@ -1,13 +1,44 @@
+import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { DropdownMenu, DropdownMenuTrigger, DropdownMenuContent } from '../../../ui/dropdown-menu';
+import PlatformCard from './PlatformCard';
+import { available_apps } from './Data';
 
 const PlatformSwitcher = () => {
     const { t } = useTranslation();
+
+    const [isDropdownOpen, setDropdownOpen] = useState(false);
+
+    const handleDropdownToggle = () => {
+        setDropdownOpen(prev => !prev);
+    };
+
     return (
-        <div className='flex cursor-pointer flex-row items-center gap-2 px-base hover:bg-disabled-100'>
-            <img src='/images/pages/header/logo_smart_trader.svg' alt='Smarttrader logo' />
-            <div className='text-base font-bold'>{t('SmartTrader')}</div>
-            <img src='/images/pages/header/ic-chevron-down.svg' alt='down arrow' />
-        </div>
+        <DropdownMenu open={isDropdownOpen} onOpenChange={handleDropdownToggle}>
+            <DropdownMenuTrigger className='hover:cursor-pointer'>
+                <div className='flex cursor-pointer flex-row items-center gap-2 px-base hover:bg-disabled-100'>
+                    <img src='/images/pages/header/logo_smart_trader.svg' alt='Smarttrader logo' />
+                    <div className='text-base font-bold'>{t('SmartTrader')}</div>
+                    <img src='/images/pages/header/ic-chevron-down.svg' alt='down arrow' />
+                </div>
+            </DropdownMenuTrigger>
+            {isDropdownOpen && (
+                <div className='fixed inset-0 top-32 hidden bg-black/[0.72] opacity-50 md:block lg:block' />
+            )}
+            <DropdownMenuContent
+                data-testid='dropdown-content'
```

Comment:
```suggestion
                data-testid='dt_dropdown_content'
```

The naming convention for test ids that we are using in the deriv-app project